pkg/provider/resource_cluster.go

```diff
@@ -597,7 +620,7 @@ func (c *clusterResource) ensureClusterIsHealthy(ctx context.Context, cluster Cl
 		})
 }
 
-func makeClusterForCreate(clusterResource ClusterResourceModel) models.Cluster {
+func (c *clusterResource) makeClusterForCreate(ctx context.Context, clusterResource ClusterResourceModel) models.Cluster {
```

Comment:
We can revert back to `func makeClusterForCreate(clusterResource ClusterResourceModel) models.Cluster` and leave it alone
and create a method to wrap around it instead. So it would be something like:
`func (c *clusterResource) makeCluster(ctx context.Context, clusterResource ClusterResourceModel) models {
cluster := makeClusterForCreate(clusterResource)
...
<your bah code>
...
}`

and then we can just use `func (c *clusterResource) makeCluster(ctx context.Context, clusterResource ClusterResourceModel) `

WDYT?